src/ops/put_object.rs

```diff
@@ -37,7 +37,7 @@ impl S3Handler for Handler {
             ctx.multipart.is_some()
         } else if ctx.req.method() == Method::PUT {
             bool_try!(ctx.path.is_object());
-            ctx.query_strings.is_none()
+            true
```

Comment:
Try this workaround
```rust
match ctx.query_strings {
    None => true,
    Some(ref qs) => qs.get("X-Amz-Signature").is_some(),
}
```

Reply:
I think this will produce behaviour inconsistent with S3. Incorrect params should match and return an access denied error.

Reply:
I'm going to figure out why E2E test failed. The routing rules may be incorrect if we set line 40 to `true`.

Reply:
Could you rebase your branch? The master branch has updated.

Reply:
Updated

Reply:
E2E test failed because the router rule of PutObject conflicts with that of UploadPart.

To solve the ambiguity, here should exclude the query parameters of UploadPart.

```rust
match ctx.query_strings {
    None => true,
    Some(ref qs) => qs.get("uploadId").is_none(),
}
```

Reply:
Good spot. Have updated with that fix